src/mixins/java/org/spongepowered/common/mixin/api/mcp/entity/player/ServerPlayerEntityMixin_API.java

```diff
@@ -447,9 +448,9 @@ public void remove() {
 
     @Override
     public boolean setLocation(final Vector3d position, final UUID world) {
-        final WorldProperties prop = Sponge.getServer().getWorldManager().getProperties(world).orElseThrow(() -> new IllegalArgumentException("Invalid World: No world found for UUID"));
-        final World loaded = Sponge.getServer().getWorldManager().loadWorld(prop).orElseThrow(() -> new IllegalArgumentException("Invalid World: Could not load world for UUID"));
-        return this.setLocation(Location.of(loaded, position));
+        final World targetWorld = Sponge.getServer().getWorldManager().getWorld(world).orElseThrow(() -> new IllegalArgumentException("Invalid World: No loaded world found for UUID"));
```

Comment:
Only final change is to not throw an exception here. Simply return false. Shouldn't bring the server down.